packages/api/src/router/user.ts

```diff
@@ -130,6 +134,125 @@ export const userRouter = createTRPCRouter({
       }
     }),
 
+  payCoinsByUserId: publicProcedure
+    .input(
+      z.object({
+        receiver: z.object({
+          id: z.string(),
+          username: z.string(),
+          avatar: z.nullable(z.string()),
+          discriminator: z.string(),
+        }),
+        sender: z.object({
+          id: z.string(),
+          username: z.string(),
+          avatar: z.nullable(z.string()),
+          discriminator: z.string(),
+        }),
+        coins: z.number(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      try {
+        /**
+         * This function set a temp thumbnail for the user
+         */
+        const tempThumbnail = setTempThumbnail(input.receiver);
+
+        /**
+         * Check in the user, have a balance for the transaction
+         */
+        const sender = await ctx.prisma.user.findUnique({
+          where: { discordId: input.sender.id },
+          select: {
+            coins: true,
+          },
+        });
+
+        /**
+         * If sender not exist, create a new user and return error tokens message
+         */
+        if (!sender) {
+          const sender = await ctx.prisma.user.create({
+            name: input.sender.username,
+            discordId: input.sender.id,
+            discordUserName: input.sender.username,
+            discordDiscriminator: input.sender.discriminator,
+            thumbnail: setTempThumbnail(input.sender),
+            coins: 0,
+          });
+
+          if (sender) {
+            return {
+              status: 'error',
+              message: 'You have no Indie Tokens for this transaction',
+            };
+          } else {
+            return {
+              status: 'error',
+              message: '500 Server error',
+            };
+          }
+        }
+
+        /**
+         * If sender, don't have coins, not aprove the transaction else,
+         * transfers coins to reciver and decrement amount in wallet sender
+         */
+        if (sender.coins <= 0) {
+          return {
+            status: 'error',
+            message: 'You have no Indie Tokens for this transaction',
+          };
+        } else {
+          /**
+           * Update coins of the sender, discount coins to pay
+           */
+          const updateSender = await ctx.prisma.user.update({
+            where: { discordId: input.sender.id },
+            data: { coins: { decrement: input.coins } },
+            select: {
+              coins: true,
+            },
+          });
+
+          /**
+           * If exist, add coins of the user receiver wallet, else
+           * create user wallet and add coins
+           */
```

Comment:
```suggestion
          /**
           * If the receiver doesn't exists, create him/her in the DB.
           * Otherwise, Increment the receiver's coins.
           */
```